### PR TITLE
Rails Capstone project: Budget App

### PR DESCRIPTION
**Pull request by benardop**
Changes I have made in this PR: 
- Splash screen
     -A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
     - The user should be able to register in the app with full name, email and password (all mandatory).
     - The user can log into the app using email and password.
     - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
     - When the user logs in, they are presented with the categories page.
     - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that 
     category.
     -  When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
     - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
     - 
- Transactions page
   - For a given category, the list of transactions is presented, ordered by the most recent.
   -  At the top of the page the user could see the total amount for the category (sum of all of the amounts of the 
      transactions in that category).
   - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
    "Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
 "Add a new transaction" page

- The user fills out a form to create a new transaction with:
  - name (mandatory)
 -  amount (mandatory)
 - categories (mandatory at least one)
 - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,168 @@
...
+- **[Display all posts]** - Display recent posts.
+- **[Delete post ]** - Delete specific post.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Version <a name="live-demo"></a>
+
+- [Comming soon]()
```
Comment (danmainah) [inline README.md:57]: - [ ] Kindly deploy your project this is project requirement 
image

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (danmainah) [inline app/models/user.rb]: - [ ] @benardop great work on your app, however, am unable to go past the login/signup page, am getting this error even when trying to create the user manually, had to remove some devise code to make it work, kindly ensure it works
image

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [ ] Kindly ensure the signup page matches the design given, check the comparison below:

Behance Design          |  Your design     
:-------------------------:|:-------------------------:
image |  image

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<h2>Log in</h2>
```
Comment (danmainah) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly ensure the signin page matches the design given, check the comparison below:

Behance Design          |  Your design     
:-------------------------:|:-------------------------:
image| image

File: app/views/budgets/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <div class="field mt-2 mb-2">
+        <%= f.text_field :name, autofocus: true, autocomplete: "name" , class: "form-control", placeholder: "Budget Name" %>
+    </div>
+    <div class="field mb-2">
+      <%= f.text_field :amount, autofocus: true, autocomplete: "name" , class: "form-control", placeholder: "Amount in numbers" %>
+    </div>
+
+    <div class="actions mb-2">
+      <%= f.submit "SAVE" , class: "btn btn-success" %>
+    </div>
```
Comment (danmainah) [inline app/views/budgets/new.html.erb:14]: - [ ] Kindly ensure the transactions items are ordered  by the most recent 
image

File: spec/requests/users_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'renders correct template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'returns correct response body' do
+      expect(response.body).to include('Users List')
+    end
+  end
+end
```
Comment (danmainah) [inline spec/requests/users_spec.rb:19]: - [ ] @benardop great work on your tests kindly add tests for the budget controller and views as done in the group and users controller

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @benardop , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️

- [ ] Kindly ensure you include a video presentation showcasing the features of the project, kindly ensure to meet all requirements below
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 7]
```diff
@@ -1,24 +1,174 @@
...
+- [x] register and log in, so that the data is private to them.
+- [x] introduce new transactions associated with a category.
+- [x] see the money spent on each category.
 
-Things you may want to cover:
+## Project Documentation 📄
 
-* Ruby version
+- [x] Here is the presentation video [link](https://www.loom.com/share/ca9b387a0a3e4f2e938d3509cfd2c9d6)👈
+<!-- - [x] Here is the deployed app in [Heroku](https://app-budget-on-rails.herokuapp.com/)👈 -->
```
Comment (Mihndim2020) [inline README.md:11]: - [OPTIONAL] As pointed out by the previous code reviewer, which is a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) for this project, " `The project should be deployed and accessible online.` ",  please deploy your app using Heroku, Netlify or GitHub Pages,
your project should be deployed properly, the home and detail page works as expected, A link to the deployed project should be present in the README.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @benardop, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 7 above)
Comment (benardop) [reply README.md:11]: @Mihndim2020 Thank you for the review. Yes I will deploy the app on Heroku as you suggested.

**Pull request merged**